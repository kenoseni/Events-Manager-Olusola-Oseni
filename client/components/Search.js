import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { history } from '../routes';

class Search extends Component {
  static propTypes = {
    // onSubmit: PropTypes.func.isRequired,
    searchVisible: PropTypes.bool
  }
  constructor(props) {
    super(props);
    this.state = {
      searchVisible: false,
      searchText: ''
    }
    this.showSearch = this.showSearch.bind(this);
    this.updateSearchInput = this.updateSearchInput.bind(this);
    this.submitForm = this.submitForm.bind(this);
  }

  updateSearchInput(e) {
    const val = e.target.value;
    this.setState({
      searchText: val
    });
  }

  // toggle visibility when run on the state
  showSearch() {
    this.setState({
      searchVisible: !this.state.searchVisible
    })
  }

  submitForm(e) {
    e.preventDefault();
    const {searchText} = this.state;
    this.setState({
      searchText: ''
    })
    this.props.searchForCenters({name:searchText});
    history.push("/search")
  }


  render() {
    
    // Classes to add to the <input /> element
    let searchInputClasses = ["searchInput, form-control mr-sm-2"];
    // Update the class array if the state is visible
    if (this.state.searchVisible) {
      searchInputClasses.push("active");
    }
    return (
      <div>
        {this.state.searchVisible &&
        <form className="form-inline" onSubmit={this.submitForm}>
          <input
            type="text"
            value={this.state.searchText}
            className={searchInputClasses.join(' ')}
            onChange={this.updateSearchInput.bind(this)}
            placeholder="Search for centers..."
          />
        </form>
        }
        {!this.state.searchVisible && <div style={{cursor: "pointer"}}
          onClick={this.showSearch}
          className="fa fa-search searchIcon">
        </div>
        }
      </div>
    );
  }
}

export default Search;
