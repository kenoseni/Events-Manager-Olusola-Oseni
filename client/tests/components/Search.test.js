import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import sinon from 'sinon';
import Connected, { AdminPanels } from '../../components/AdminPanels';
import Search from '../../components/Search';

configure({ adapter: new Adapter() });
const mockupdateSearchInput = jest.fn();
const mocksubmitForm = jest.fn();
const mocksearchForCenters = jest.fn();
describe('<Search />', () => {
  let props = {
    path: '/',
    history: {
      push: '/'
    }
  };
  const e = {
    preventDefault: jest.fn(),
    target: {
      value: ''
    }
  };
  const wrapper = shallow(
    <Search searchForCenters={mocksearchForCenters} {...props} />
  );
  it('should render a <Search /> component', () => {
    expect(wrapper).toMatchSnapshot();
  });
  // it('should call the submitform function', () => {
  //   const spy = sinon.spy(wrapper.instance(), 'submitForm');
  //   wrapper.instance().submitForm(e);
  //   expect(mocksubmitForm).toHaveBeenCalledTimes(1);
  // });
});
