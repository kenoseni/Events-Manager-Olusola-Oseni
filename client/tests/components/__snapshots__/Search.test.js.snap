// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> should render a <Search /> component 1`] = `
<div>
  <div
    className="fa fa-search searchIcon"
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
      }
    }
  />
</div>
`;
